edge-util/src/Microsoft.Azure.Devices.Edge.Util/Metrics.cs

```diff
@@ -85,14 +85,49 @@ static void StartReporting(IMetricsRoot metricsCollector)
             scheduler.Start();
         }
 
-        public static void Count(MetricTags tags, CounterOptions options)
+        public static void CountIncrement(MetricTags tags, CounterOptions options, long amount)
         {
             Preconditions.CheckNotNull(tags);
             Preconditions.CheckNotNull(options);
+            Preconditions.CheckNotNull(amount);
 
             MetricsCollector.ForEach(mroot =>
             {
-                mroot.Measure.Counter.Increment(options, tags);
+                mroot.Measure.Counter.Increment(options, tags, amount);
+            });
+        }
+
+        public static void CountDecrement(MetricTags tags, CounterOptions options, long amount)
+        {
+            Preconditions.CheckNotNull(tags);
+            Preconditions.CheckNotNull(options);
+            Preconditions.CheckNotNull(amount);
+
+            MetricsCollector.ForEach(mroot =>
+            {
+                mroot.Measure.Counter.Decrement(options, tags, amount);
+            });
+        }
+
+        public static void CountIncrement(CounterOptions options, long amount)
+        {
+            Preconditions.CheckNotNull(options);
+            Preconditions.CheckNotNull(amount);
+
+            MetricsCollector.ForEach(mroot =>
+            {
+                mroot.Measure.Counter.Increment(options, amount);
+            });
+        }
+
+        public static void CountDecrement(CounterOptions options, long amount)
+        {
+            Preconditions.CheckNotNull(options);
+            Preconditions.CheckNotNull(amount);
+
+            MetricsCollector.ForEach(mroot =>
+            {
+                mroot.Measure.Counter.Decrement(options, amount);
```

Comment:
Does this set tags to null? If so, can we just call the other overload?

Reply:
I'm not sure what its doing under the covers. I'd rather keep it the way it is to prevent passing in null explicitly.